grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/cfg/GrailsDomainBinder.java

```diff
@@ -1152,7 +1152,12 @@ protected String calculateTableForMany(ToMany property, String sessionFactoryBea
             return addUnderscore(left, propertyColumnName);
         }
 
-        String right = getTableName(property.getAssociatedEntity(), sessionFactoryBeanName);
+        String right;
+        try {
+            right = getTableName(property.getAssociatedEntity(), sessionFactoryBeanName);
```

Comment:
Instead of catching the exception, check if `property.getAssociatedEntity()` is null, then throw an error